File: test/fixtures/vulnerable.php
<?php
// Deliberately vulnerable PHP fixture.

// VULN: SQL Injection — concatenation into mysqli_query()
function get_user($conn) {
    $id = $_GET['id'];
    return mysqli_query($conn, "SELECT * FROM users WHERE id = " . $id);
}

// VULN: Command Injection — shell_exec with untrusted input
function ping() {
    $host = $_POST['host'];
    return shell_exec("ping -c 1 " . $host);
}

// VULN: Insecure Crypto — MD5
function digest($data) {
    return md5($data);
}

// VULN: Hardcoded Secret
$password = "s3cr3t-production-value";

File: test/fixtures/vulnerable.go
// Deliberately vulnerable Go fixture.
package main

import (
	"database/sql"
	"net/http"
	"os/exec"
)

// VULN: SQL Injection — concatenation into Query()
func getUser(db *sql.DB, r *http.Request) (*sql.Rows, error) {
	id := r.URL.Query().Get("id")
	return db.Query("SELECT * FROM users WHERE id = " + id)
}

// VULN: Command Injection — shell invocation with concatenated input
func ping(r *http.Request) error {
	host := r.FormValue("host")
	return exec.Command("sh", "-c", "ping -c 1 "+host).Run()
}
